### PR TITLE
Jake/add multiple loss functions

### PR DESCRIPTION
**Pull request by jake-ringdna**
These changes allow for scoring methods other than F1. The impetus for this change is that we're really trying to optimize recall for the current demo set model. In future models, we may want to maximize precision, accuracy, etc. DAGs have been separately adjusted to pass "recall" as the optimization metric. Branch name references "loss function", specifically because of the way values are negated (Hyperopt seeks to minimize outputs).

### CODE REVIEW

**Comment by jake-ringdna** [comment]
Hi @correlator, thanks for reviewing! The main issue is that all of our inaccuracy is coming from type II error, and type I is more useful for the current model. Therefore, instead of optimizing for F1, I'm currently optimizing for recall, but we could also optimize for precision, etc. depending on what model we use. It also opens up some options for non-binary classifiers. All of the options are listed [here](https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter).

**Comment by correlator** [comment]
Very nice.  Might be worth adding a comment or docstring that links to the list of options.  That will make future onboarding more smooth.
